Terminal.Gui/Views/NumericUpDown.cs

```diff
@@ -0,0 +1,224 @@
+                        {
+                            return false;
+                        }
+
+                        if (Value is { })
+                        {
+                            Value = (dynamic)Value + (dynamic)Increment;
+                        }
+
+                        return true;
+                    });
+
+        AddCommand (
+                    Command.ScrollDown,
+                    () =>
+                    {
+                        if (type == typeof (object))
+                        {
+                            return false;
+                        }
+
+                        if (Value is { })
+                        {
+                            Value = (dynamic)Value - (dynamic)Increment;
+                        }
+
+                        return true;
+                    });
+
+        KeyBindings.Add (Key.CursorUp, Command.ScrollUp);
+        KeyBindings.Add (Key.CursorDown, Command.ScrollDown);
+
+        SetText ();
+
+        return;
+
+        void OnDownButtonOnAccept (object? s, HandledEventArgs e) { InvokeCommand (Command.ScrollDown); }
+
+        void OnUpButtonOnAccept (object? s, HandledEventArgs e) { InvokeCommand (Command.ScrollUp); }
+    }
+
+    private T _value = default!;
+
+    /// <summary>
+    ///     Gets or sets the value that will be incremented or decremented.
+    /// </summary>
+    /// <remarks>
+    ///     <para>
+    ///         <see cref="ValueChanging"/> and <see cref="ValueChanged"/> events are raised when the value changes.
+    ///         The <see cref="ValueChanging"/> event can be canceled the change setting <see cref="CancelEventArgs{T}.Cancel"/> to
+    ///         <see langword="true"/>.
+    ///     </para>
+    /// </remarks>
+    public T Value
+    {
+        get => _value;
+        set
+        {
+            if (_value.Equals (value))
+            {
+                return;
+            }
+
+            T oldValue = value;
+            CancelEventArgs<T> args = new (in _value, ref value);
+            ValueChanging?.Invoke (this, args);
+
+            if (args.Cancel)
+            {
+                return;
+            }
+
+            _value = value;
+            SetText ();
+            ValueChanged?.Invoke (this, new (in value));
+        }
+    }
+
+    private string _format = "{0}";
+
+    /// <summary>
+    ///     Gets or sets the format string used to display the value. The default is "{0}".
+    /// </summary>
+    public string Format
+    {
+        get => _format;
+        set
+        {
+            if (_format == value)
+            {
+                return;
+            }
+
+            _format = value;
+            SetText ();
+        }
+    }
+
+    private void SetText ()
+    {
+        _number.Text = string.Format (Format, _value);
+        Text = _number.Text;
+    }
+
+    /// <summary>
+    ///     Raised when the value is about to change. Set <see cref="CancelEventArgs{T}.Cancel"/> to true to prevent the
```

Comment:
Remove the `Cancel` property because it's only exposed an a instance.

Reply:
I don't know what you mean.

Reply:
The cref="CancelEventArgs{T}.Cancel"/> is generation a warning and remove it the warning disappear.